kasa/modules/rulemodule.py

```diff
@@ -29,30 +30,38 @@ class TimeOption(Enum):
 class Rule(BaseModel):
     """Representation of a rule."""
 
-    id: str
+    # not used when adding a rule
+    id: Optional[str]
     name: str
-    enable: bool
+    enable: int
     wday: List[int]
-    repeat: bool
+    repeat: int
```

Comment:
I think we should let pydantic to perform conversion to bool here, or is there some reason not to?

Reply:
The device need to receive a 0 or 1 value in the json not a true or false, Fixed in https://github.com/python-kasa/python-kasa/pull/501/commits/cb5722faae4641b4e61a9a89cc37612fc61559e3 by adding a `EnabledOption` class

Reply:
Hmm, I think pydantic should handle that conversion just fine by defining a custom serializer, I hope at least. It would be better to avoid custom boolean types to keep the API simple.